pkg/common-controller/groupsnapshot_controller_helper.go

```diff
@@ -590,24 +594,26 @@ func (ctrl *csiSnapshotCommonController) createSnapshotsForGroupSnapshotContent(
 			metav1.SetMetaDataAnnotation(&volumeSnapshotContent.ObjectMeta, utils.AnnDeletionSecretRefNamespace, groupSnapshotSecret.Namespace)
 		}
 
-		label := make(map[string]string)
-		label[utils.VolumeGroupSnapshotNameLabel] = groupSnapshotContent.Spec.VolumeGroupSnapshotRef.Name
 		volumeSnapshot := &crdv1.VolumeSnapshot{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:       volumeSnapshotName,
 				Namespace:  volumeSnapshotNamespace,
 				Labels:     label,
 				Finalizers: []string{utils.VolumeSnapshotInGroupFinalizer},
 			},
-			Spec: crdv1.VolumeSnapshotSpec{
-				Source: crdv1.VolumeSnapshotSource{
-					PersistentVolumeClaimName: &pv.Spec.ClaimRef.Name,
-				},
-			},
+			// The spec stanza is set immediately
 			// The status will be set by VolumeSnapshot reconciler
 		}
 
-		_, err = ctrl.clientset.SnapshotV1().VolumeSnapshotContents().Create(ctx, volumeSnapshotContent, metav1.CreateOptions{})
+		// If we didn't found any persistent volume, we revert back to a static-provisioning
```

Comment:
I'm not sure if this is right.  If PV not found, it could be that it got deleted unexpectedly. I think this is not a reliable way to determine whether it is static or dynamic provisioning.

Reply:
As discussed, for static provisioning, we'll let the admin create individual VolumeSnapshot and VolumeSnapshotContent objects.

Reply:
That was meant to prevent the snapshot controller from panicking when the PV is not found.
This code will be used only for dynamic provisioning.